Style cyborg system.

mcglincy opened the pull request:
Kevin B worked through the styling on a codepen site, and then I turned it into an exlibris PR.

I've been using puma-dev to start my local server so I can access the subdomain site URLs. Should I just update Procfile.dev or do we want both a Procfile.dev and a Procfile.puma-dev?

I also needed to do a "bundle lock --add-platform x86_64-darwin" for my local machine (intel macbook).

mcglincy commented:
Test failure in set_current_request_details_test.rb when cy_borg system is set to live: true in systems.yml.

```
      context "as an anonymous visitor" do
        should "raise ActiveRecord::RecordNotFound" do
          assert_raises(ActiveRecord::RecordNotFound) do
            get system_url(subdomain: "cyborg")
          end
        end
      end
```

@jdbann Should I leave cy_borg as live: false, add another live: false system for this RecordNotFound test, or something else?

brocktopus commented:
Dumb question, but any chance it might be a spelling issue (needing "cy_borg" instead of "cyborg" or the like)?

jdbann commented:
> @jdbann Should I leave cy_borg as live: false, add another live: false system for this RecordNotFound test, or something else?

Probably worth adding a new one @mcglincy - but maybe we call it something like `not_live` so the purpose is clear and it doesn't collide in future?

mcglincy commented:
> > @jdbann Should I leave cy_borg as live: false, add another live: false system for this RecordNotFound test, or something else?
> 
> Probably worth adding a new one @mcglincy - but maybe we call it something like `not_live` so the purpose is clear and it doesn't collide in future?

Added a not_live system for the test case, and fixed a couple of stylelint issues.

mcglincy commented:
> Dumb question, but any chance it might be a spelling issue (needing "cy_borg" instead of "cyborg" or the like)?

The test passes/fails if cy_borg is set to live false/true, so the test is system liveness dependent. I've created a new for-testing system that we can leave as live: false for this particular unit test.

jdbann commented on app/assets/stylesheets/themes/cyborg.css:
```diff
@@ -0,0 +1,71 @@
+/* TODO: debug why _head.html+cyborg.erb isn't loading fonts right */
```
What's the issue here?

mcglincy replied:
This was before I found _typography.scss :). I'll remove the extraneous comment.

jdbann commented:
> I've been using puma-dev to start my local server so I can access the subdomain site URLs. Should I just update Procfile.dev or do we want both a Procfile.dev and a Procfile.puma-dev?

Interesting question... This app is pretty dependent on being able to run subdomains so I think it's reasonable to bake `puma-dev` into the experience. When I follow the [installation instructions](https://github.com/puma/puma-dev#macos-support) though, it runs `puma-dev` as a background process - so it's always available at `exlibrisrpg.test` without the need to start the server.

Have you got it setup some other way? I'd be open to changing this so that `puma-dev` doesn't have to be running in the background all the time.

mcglincy commented:
> Have you got it setup some other way? I'd be open to changing this so that `puma-dev` doesn't have to be running in the background all the time.

I don't have puma-dev running in the background, and instead I've been manually starting / stopping it on the command line (via "foreman start -f Procfile.puma-dev", with the Prodfile.puma-dev file on this PR) and letting puma-dev use its default unprivileged port (9280). I sometimes run an externally visible server on my dev machine, so avoiding 80/443 for puma-dev is more convenient for me.

mcglincy commented:
@jdbann I rolled back Gemfile.lock and Procfile-related changes from this PR, as they seem more appropriate for a separate PR and discussion.

Do you have cycles to review & merge this PR before the holiday, so the cyborg system can be ready to launch? Thanks!

jdbann reviewed (approved):
@mcglincy Sorry it has taken so long for me to get back to you on this. I'm happy to put this in as it is.

Would you be confident joining as a maintainer on this project so I'm not a blocker for this kind of thing in future?

mcglincy commented:
Thanks for the approval! 

And sure, I'd be happy to be a maintainer. I am very much not a Rails or Tailwind expert, but I can help unblock simpler changes and will defer anything deeper or more complicated to you.
